qiskit/visualization/gate_map.py

```diff
@@ -1051,128 +1033,87 @@ def plot_coupling_map(
-                    y_arrow = -y_start - dy * 0.5
-                    dx_arrow = dx * 0.2
-                    dy_arrow = -dy * 0.2
-                    head_width = 0.2
-                ax.add_patch(
-                    mpatches.FancyArrow(
-                        x_arrow,
-                        y_arrow,
-                        dx_arrow,
-                        dy_arrow,
-                        head_width=head_width,
-                        length_includes_head=True,
-                        edgecolor=None,
-                        linewidth=0,
-                        facecolor=line_color[ind],
-                        zorder=1,
-                    )
-                )
-
-    # Add circles for qubits
-    for var, idx in enumerate(grid_data):
-        _idx = [idx[1], -idx[0]]
-        ax.add_artist(
-            mpatches.Ellipse(
-                _idx,
-                qubit_size / 48,
-                qubit_size / 48,  # This is here so that the changes
-                color=qubit_color[var],
-                zorder=1,
-            )
-        )  # to how qubits are plotted does
-        if label_qubits:  # not affect qubit size kwarg.
-            ax.text(
-                *_idx,
-                s=qubit_labels[var],
-                horizontalalignment="center",
-                verticalalignment="center",
-                color=font_color,
-                size=font_size,
-                weight="bold",
-            )
-    ax.set_xlim([-1, x_max + 1])
-    ax.set_ylim([-(y_max + 1), 1])
-    ax.set_aspect("equal")
+    graph = CouplingMap(coupling_map).graph
+
+    if not plot_directed:
+        graph = graph.to_undirected()
+
+    for node in graph.node_indices():
+        graph[node] = node
+
+    for edge_index in graph.edge_indices():
+        graph.update_edge_by_index(edge_index, edge_index)
+
+    # pixel-to-inch conversion
+    px = 1 / plt.rcParams["figure.dpi"]
+
+    if qubit_coordinates:
+        qubit_coordinates = [coordinates[::-1] for coordinates in qubit_coordinates]
+
+    def color_node(node):
+        if qubit_coordinates:
+            out_dict = {
+                "label": str(qubit_labels[node]),
+                "color": f'"{qubit_color[node]}"',
+                "fillcolor": f'"{qubit_color[node]}"',
+                "style": "filled",
+                "shape": "circle",
+                "pos": f'"{qubit_coordinates[node][0]},{qubit_coordinates[node][1]}"',
+                "pin": "True",
+            }
+        else:
+            out_dict = {
+                "label": str(qubit_labels[node]),
+                "color": f'"{qubit_color[node]}"',
+                "fillcolor": f'"{qubit_color[node]}"',
+                "style": "filled",
+                "shape": "circle",
+            }
+        out_dict["fontcolor"] = f'"{font_color}"'
+        out_dict["fontsize"] = str(font_size)
+        out_dict["height"] = str(qubit_size * px)
+        out_dict["pin"] = "True"
```

Comment:
I feel like this isn't needed here, since it's done in the `if qubit_coordinates` case above, but it also doesn't hurt since `pin` doesn't have any meaning unless we set `pos` according to https://graphviz.org/docs/attrs/pin/

```suggestion
```